### PR TITLE
🎉Source Postgres: Set up connection - add schema selection

### LINKED ISSUE #1435: Source Postgres: Set up connection - add schema selection
## Tell us about the problem you're trying to solve
I tried to move data from Postgres DB, now I have several Schemas and wanted to move data from only one schema.
If I want to do it currently, I need to select the required tables one by one

## Describe the solution you’d like
mark the schema that I want to load, by that the checkmark of the tables would be updated

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368203873756) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by etsybaev**
## What
There is no way to select particular schemas on UI. So the "discover" method may work pretty slow or fail at all in case we have lots of streams in db..

## How
Added a new optional field to select a list of schemas to process. This should prevent some crashes when we have lots of schemas and lot's of streams in each. 

Also checked on UI:
Selection_356
Selection_357
Selection_358

DB contains much more schemas (including pretty big ones) then we synced 
Selection_359
Sync result:
Selection_360

Also tested, that connector still works if no schema is selected explicitly on UI.
Selection_369
Selection_371
Selection_372
Selection_373

## Recommended reading order
1. spec.json
2. PostgresSource.java 

## 🚨 User Impact 🚨
One more field has been added to UI, but it's not mandatory

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [x] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [x] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json [hunk 1]
```diff
@@ -29,31 +29,42 @@
         "type": "string",
         "order": 2
       },
+      "schemas": {
+        "title": "Schemas",
+        "description": "The list of schemas to sync from. Defaults to user. Case sensitive.",
```
Comment (sashaNeshcheret) [inline airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:34]: Wdyt about adding info about comma separator if we use more than 1 schema, or it is obvious?

**Review by sashaNeshcheret — APPROVED** [review]

**Review by VitaliiMaltsev — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json (hunk 1 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:34]: Good question. Please check the first screenshot. The UI represents it as different items but sends it to the backend as an array. So no explicit separators are required

File: airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Postgres Source Spec",
     "type": "object",
-    "required": ["host", "port", "database", "username"],
+    "required": ["host", "port", "database", "schemas", "username"],
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:7]: If we make this new field required, all existing Postgres connection config will become invalid. Can we make it optional, and keep the current behavior to retrieve all schemas if this parameter is missing?

File: airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json [hunk 3]
```diff
@@ -29,31 +29,42 @@
...
+      "schemas": {
+        "title": "Schemas",
+        "description": "The list of schemas to sync from. Defaults to user. Case sensitive.",
+        "type": "array",
+        "items": {
+          "type": "string"
+        },
+        "minItems": 1,
+        "uniqueItems": true,
+        "order": 3
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:41]: The most common Postgres schema is `public`, right? If that's the case, we can add a default value here.

```
"default": ["public"]
```

**Review by tuliren — CHANGES_REQUESTED** [review]
This is great! However, I think it's better to keep the config backward-compatible, especially because Postgres source is pretty popular. Otherwise, this will cause quite some cloud on-call work from existing Postgres connections.

File: airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json (hunk 2 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json]: Updated. Added one more test to check that we may keep this field empty and also tested on UI manually (see attached captures in PR's description). Thanks

File: airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json (hunk 3 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-postgres/src/main/resources/spec.json:42]: done, thanks

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java [hunk 4]
```diff
@@ -130,17 +130,21 @@ public AirbyteCatalog discover(final JsonNode config) throws Exception {
...
-      LOGGER.debug("Found table: {}", info.getName());
+    if (schemas != null && !schemas.isEmpty()) {
+      // process explicitly selected (from UI) schemas
+      final List<TableInfo<CommonField<JDBCType>>> internals = new ArrayList<>();
+      for (String schema : schemas) {
+        LOGGER.debug("Discovering schema: {}", schema);
+        internals.addAll(super.discoverInternal(database, schema));
+      }
+      for (TableInfo<CommonField<JDBCType>> info : internals) {
+        LOGGER.debug("Found table: {}", info.getName());
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:141]: Nitpick. Some tables may have the same names, but under different schemas. If we are going to log them, I think it's better to log the schema as well:

```suggestion
        LOGGER.debug("Found table (schema: {}): {}", info.getNameSpace(), info.getName());
```

**Review by tuliren — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java (hunk 4 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java]: Updated, thanks
